src/modules/Dashboard/features/Applet/Activities/Activities.tsx

```diff
@@ -1,18 +1,266 @@
+    [actions, appletId],
+  );
+
+  const activities = useMemo(
+    () => (activitiesData?.result ?? []).map((activity) => formatRow(activity)),
+    [activitiesData, formatRow],
+  );
 
   useEffect(() => {
-    // TODO: https://mindlogger.atlassian.net/browse/M2-5443
-    // This is just a placeholder until we implement the activities view
-    window.setTimeout(() => {
-      setIsLoading(false);
-    }, 1000);
-  }, []);
-
-  return <StyledBody>{isLoading ? <Spinner /> : null}</StyledBody>;
+    if (!appletId) return;
+
+    handleReload();
+
+    return () => {
+      setActivitiesData(null);
+    };
+  }, [appletId]);
+
+  const isEmpty = !activities?.length && !isLoading;
+
+  const getEmptyComponent = () => {
+    if (isEmpty) {
+      if (searchValue) {
+        return t('noMatchWasFound', { searchValue });
+      }
+
+      return appletId ? t('noActivitiesForApplet') : t('noActivities');
+    }
+  };
+
+  return (
+    <StyledBody>
+      {isLoading && <Spinner />}
+
+      <StyledFlexWrap sx={{ gap: 1.2, mb: 2.4 }}>
+        {appletId && (
+          <>
+            <StyledFlexTopCenter sx={{ gap: 1.2 }}>
+              <StyledButton
+                variant="outlined"
+                startIcon={<Svg width={18} height={18} id="slider-rows" />}
+                // TODO: Implement filters
+                // https://mindlogger.atlassian.net/browse/M2-5530
+                onClick={() => alert('TODO: filters')}
+                data-testid={`${dataTestid}-filters`}
+              >
+                {t('filters')}
+              </StyledButton>
+
+              <ButtonWithMenu
+                variant="outlined"
+                label={t('sortBy')}
+                anchorEl={anchorEl}
+                setAnchorEl={setAnchorEl}
+                // TODO: Implement sorting
+                // https://mindlogger.atlassian.net/browse/M2-5445
+                menuItems={[
+                  {
+                    title: 'TODO: Sort options',
+                    action: () => alert('TODO: Sort options'),
+                  },
+                ]}
+                startIcon={<></>}
+                data-testid={`${dataTestid}-sort-by`}
+              />
+            </StyledFlexTopCenter>
+
+            <StyledFlexWrap sx={{ gap: 1.2, ml: 'auto' }}>
+              <StyledSearch
+                withDebounce
+                // TODO: Implement search
+                // https://mindlogger.atlassian.net/browse/MLG-21
+                placeholder={`TODO: ${t('searchActivities')}`}
+                onSearch={handleSearch}
+                data-testid={`${dataTestid}-search`}
+              />
+
+              <StyledButton
+                variant="outlined"
+                // TODO: Implement assign
+                // https://mindlogger.atlassian.net/browse/M2-5710
+                onClick={() => alert('TODO: Assign activity')}
+                data-testid={`${dataTestid}-assign`}
+                sx={{ minWidth: '10rem' }}
+              >
```

Comment:
Since this is a phase 2 feature, I consider this low priority. However, we may want to create custom styles for this button. It's no longer an outlined button according to the [latest design](https://www.figma.com/file/6BNx63peNVzrHAF1FoZ78x/Multi-Informant-2024?type=design&node-id=1558-131390&mode=design&t=9hYvaQNBGNW3fIV1-4), but there's no contained version of this component that uses the `secondary_container` color

```typescript
export const StyledAssignButton = styled(Button)`
    background-color: ${variables.palette.secondary_container};
    color: ${variables.palette.on_secondary_container};
    font-size: ${variables.font.weight.regular};
`;
```

Reply:
Ah yes, I meant to call this out and forgot. I did start trying to theme this new button variant, but ended up going down a bit of a rabbit hole trying to decide the best way to add it to the theme config, thinking I'd need to add a new `color` style (rather than `variant`, which has limited options), which would also require some conditional logic to adjust font weight based on that, and dealing with hover/active states, etc. I was reticent to add one-off styling that lives in this screen's `.styles.ts` file, since this variant will inevitably be used elsewhere, and I'd like to avoid duplicate/stale code.

Maybe best to create a separate task for this change (there may be other missing button variants, too) and mention it here as being out of scope for this task.

Reply:
Works for me. Be sure to link it to [M2-5710](https://mindlogger.atlassian.net/browse/M2-5710) when you create the ticket

Reply:
Created [M2-6071](https://mindlogger.atlassian.net/browse/M2-6071) to handle this.